ws-client/src/manager.rs

```diff
@@ -34,32 +34,60 @@ pub enum RequestStatus {
 	Invalid,
 }
 
-type PendingCallOneshot = oneshot::Sender<Result<JsonValue, Error>>;
+type PendingCallOneshot = Option<oneshot::Sender<Result<JsonValue, Error>>>;
 type PendingSubscriptionOneshot = oneshot::Sender<Result<(mpsc::Receiver<JsonValue>, SubscriptionId), Error>>;
 type SubscriptionSink = mpsc::Sender<JsonValue>;
 type UnsubscribeMethod = String;
-type RequestId = u64;
+type RequestId = u8;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 /// Manages and monitors JSONRPC v2 method calls and subscriptions.
 pub struct RequestManager {
+	/// Free list, 0 indicates a free slot and 1 an occupied slot.
+	free_list: [u8; 256],
```

Comment:
Suggestion 1: rename `free_list` to `request_id_slots`.

Suggestion 2: Change the type to `[bool; 256]` or `[Slot; 256]` where `Slot` is an enum with `Vacant` and `Occupied` or some such to make the state clearer.

Could also make it dense with `[u64; 4]` for 1/8th of memory, and using [`leading_zeroes`](https://doc.rust-lang.org/std/primitive.u64.html#method.leading_zeros) to make the iteration technically faster, at least on some implementations of x86.

Reply:
I made it an `VecDeque` instead and made it possible configure the size, ok? However, I didn't implement it in the client though I couldn't think of a good name and the config is already complicated....

Feel free to push a fix to this branch if you have something better in mind :)